src/redux/reducers/ws/execution.js

```diff
@@ -119,16 +115,23 @@ function reducer(state = getInitialState(), action = {}) {
     }
 
     case types.SET_LIVE_EXECUTION_TRADES: {
-      const { positionData } = payload
+      const { positionData, strategyMapKey, isOpened } = payload
       const { id } = positionData
-      const currentState = state.results?.positions || {}
+      const positions = state.results[strategyMapKey]?.strategy[
+        isOpened ? 'openPositions' : 'closedPositions'
+      ]
+
+      const currentPosition = { ...positions[id] = {}, ...positionData }
+
       const newState = {
         ...state.results,
-        positions: {
-          ...currentState,
-          [id]: {
-            ...(currentState?.[id] || {}),
-            ...positionData,
+        [strategyMapKey]: {
+          ...state.results[strategyMapKey],
+          strategy: {
+            [isOpened ? 'openPositions' : 'closedPositions']: {
+              ...positions,
+              [id]: currentPosition,
+            },
           },
```

Comment:
- can defined var for state.results[strategyMapKey]
- spread is missed for strategy, otherwise it will remove closedPositions when updating state for openPositions

`          strategy: {
            ...state.results[strategyMapKey].strategy
            [isOpened ? 'openPositions' : 'closedPositions']: {
              ...positions,
              [id]: currentPosition,
            },
          },`